go/ql/src/Security/CWE-601/OpenUrlRedirectGood.go

```diff
@@ -3,21 +3,26 @@ package main
 import (
 	"net/http"
 	"net/url"
+	"strings"
 )
 
-func serve() {
+func serve1() {
 	http.HandleFunc("/redir", func(w http.ResponseWriter, r *http.Request) {
 		r.ParseForm()
-		target, err := url.Parse(r.Form.Get("target"))
+		targetUrl := r.Form.Get("target")
+		// replace all backslashes with forward slashes before parsing the URL
```

Comment:
I think the query doesn't really require this, the hostname check below is enough for the alert to go away. Would this cause confusion among users if they expect the alert to pop up when this replacement doesn't happen?

Reply:
It might, but I thought it better to show a solution that covers all bases.